components/style/properties/helpers/animated_properties.mako.rs

```diff
@@ -622,6 +623,16 @@ impl<T: RepeatableListInterpolate> Interpolate for Vec<T> {
     }
 }
 
+impl<T: RepeatableListInterpolate> Interpolate for SmallVec<[T; 1]> {
+    fn interpolate(&self, other: &Self, progress: f64) -> Result<Self, ()> {
+        use num_integer::lcm;
+        let len = lcm(self.len(), other.len());
+        self.iter().cycle().zip(other.iter().cycle()).take(len).map(|(me, you)| {
+            me.interpolate(you, progress)
+        }).collect()
+    }
```

Comment:
Not wild about this duplication. Can you implement this for anything that's VecLike + has an iter() method? VecLike should really require iter() IMO, but making a breaking change to smallvec would be annoying, so you can just add a trait here.